lib/rules/prefer-find-by.ts

```diff
@@ -373,6 +373,10 @@ export default createTestingLibraryRule<Options, MessageIds>({
           const callArguments = getQueryArguments(argument.body);
           const queryMethod = fullQueryMethod.split('By')[1];
 
+          if (!queryMethod) {
+            return;
```

Comment:
This partially fixes the issue. This `return` is wrapped within `isScreenSyncQuery` condition, so it will be only applied to `screen.foo` scenarios.

You need to do the same check + `return` a few lines below here, where the other `fullQueryMethod.split('By')[1]` is located.

Reply:
Thanks for catching that! I missed that this would not work for `container` queries.

Reply:
Hmm actually, I've added the test however it is not failing. `isScreenSyncQuery` is returning true.

```js
function isScreenSyncQuery(node: TSESTree.ArrowFunctionExpression) {
...
  return (
    helpers.isSyncQuery(node.body.callee.property) ||
    isWrappedInPresenceAssert || // this is evaluating to true
    isWrappedInNegatedPresenceAssert
  );
}

Reply:
Ooh, you are right! My bad, sorry: I assumed the isScreenSyncQuery was coming from our helpers, but it's a function from the rule itself. This is fine then!